### DISCUSSION TITLE
Obtainium installation guide?

### DISCUSSION THREAD

**Discussion by bismurphy** [ImranR98/Obtainium#1266] [General]
Hello,

I am interested in trying Obtainium, but I'm finding the documentation for installation somewhat lacking. From the repo's README, there is a heading called "Installation", but all it contains is a link to the Releases page to download the files.

My main confusion here revolves around what to do with the files. It looks like each release version has 18 files associated. I believe the relevant file for installation will be one of `app-arm64-v8a-release.apk`, `app-armeabi-v7a-release.apk`, `app-release.apk`, or `app-x86_64-release.apk`, but there is nothing to indicate to me which one I should be using, or how to install an APK onto my phone. It would be greatly appreciated if the installation steps could be more directly listed as a procedure that users can follow. 

In the meantime, any guidance in this thread for me specifically would be nice. Particularly, I have a Moto G Power (Model number XT2041-4) and I'm not clear on whether I need to download the file by opening Github on my phone's browser, or if I need to download it to my computer (running Linux Mint 21) and then install it onto the phone via USB.

Thank you for your time.

**Comment by DarkCrypt** [comment]
Downloading and installing an apk on mobile is basically the same process as you would do on PC.

1. Find an apk online. (From trusted sources)
2. Click and download.
3. Click and install from download folder or browser. (Installing apps is usually handled by the internal Package installer)
4. Delete downloaded apk file after install. (Saves device storage space)

*If you've never done this before, you might have to grant your file manager "All files access" permission.

The confusion to know which apk to download is understandable. Unfortunately, this falls on the users knowledge of their specific device. If your device doesn't say somewhere within its Settings you can find this info by using 3rd party utility/tool apps or by researching online.

A decent place for online mobile research is [phonedb.net](https://phonedb.net/) which I'll reference to below.

Moto G Power XT2041-4:
[Device](https://phonedb.net/index.php?m=device&id=16430&c=motorola_moto_g_power_2020_td-lte_na_64gb_xt2041-4__g8_power__motorola_sofia) - Specs.
[CPU](https://phonedb.net/index.php?m=processor&id=790&c=qualcomm_snapdragon_665_sm6125__trinket) - Specs. (We're looking for -> Supported Instruction Set(s): ARMv8-A (A32, A64))

ARMv8-A (32bit or 64bit) is your main target architecture but your device does support others as well which are not listed. Otherwise, if your not sure which architecture you should install you can usually install the "universal" apk or `app-release.apk` in Obtainium's case which supports most devices. (The "universal" apk will usually be a larger file size)

*When using Obtainium to install apps, Obtainium will tell you what architectures your device supports if using the selector to pick an apk to install. More advanced users won't use this selector.

**Reply by Nichtraucher** [reply]
> The confusion to know which apk to download is understandable. Unfortunately, this falls on the users knowledge of their specific device.

When I'm installing apps from within AuroraStore, I don't have to make a choice and in NeoStore, one version is highlighted as "recommended". I suspect that this is not possible in Obtainium because Obtainium is technically not an app store app (and sites like github aren't app stores), correct?

**Reply by DwainZwerg** [reply]
With F-Droid/NeoStore, the NeoStore makes this selection automatically (Obtainium also tries to do this when installed; but since Obtainium can download any APKs from any source and the naming of the APKs is not standardized in any way (F-Droid has the advantage that it is standardized between the individual F-Droid repositories) and not everyone would comply with such a standard, the user often has to manually intervene).

The Google Play Store does not store APKs, but AABs. They work more like XAPKs: There is one APK file in the AAB that contains the app, one APK per configuration for the screen diagonal, one APK per configuration for the resolution and one APK per configuration for the arch. The Google Play Store downloads this AAB and extracts the app and the appropriate configurations for your respective device and then installs them in a specific order. The AuroraStore downloads the appropriate configurations and the app for your device and also installs the APKs in a specific order.
